### PR TITLE
Update kservice.c

### PR DESCRIPTION
**Pull request by CuttySark1869**
For compatibility with 16bit addressing 32bit CPU (e.g. Ti C28x).

## 拉取/合并请求描述：(PR description)

[
1. 修改了rt_memset 使其 word size independent.
2. 修改了rt_vsnprintf 使其适应于 16bit 和 32bit stack alignment. This is a bit tricking by using "va_arg(args, int)".
3. Further discussion is needed on whether or nor shall we create separate memory-related service functions for 16-bit addressing CPUs. From my point of view, I think rt_memset can be kept like this (independent of addressing mode), but rt_vsnprintf should be dealt with carefully since va_arg(args, int) may corrupt the argument passing mechanism if the size of int is not the default stack alignment size.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [x] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/contribution_guide/coding_style_en.txt)

### CODE REVIEW

**Comment by mysterywolf** [comment]
related: https://github.com/RT-Thread/rt-thread/pull/6254

File: src/kservice.c [hunk 1]
```diff
@@ -169,14 +170,15 @@ RT_WEAK void *rt_memset(void *s, int c, rt_ubase_t count)
...
         {
-            buffer = (d << 8) | d;
-            buffer |= (buffer << 16);
+            *(((unsigned char *)&buffer)+0) = d;
+            *(((unsigned char *)&buffer)+1) = d;
+            *(((unsigned char *)&buffer)+2) = d;
+            *(((unsigned char *)&buffer)+3) = d;
         }
-        else
+        else if (LBLOCKSIZE == 2)
```
Comment (Guozhanxin) [inline src/kservice.c:178]: 这个只有2和4两种情况吗？有没有可能是8？

Comment (CuttySark1869) [reply src/kservice.c:178]: 之前的代码只区分了2和4两种情况，这里是继承下来的。LBLOCKSIZE = sizeof(long)。一般long=int=32bit，64位的整型一般是long long。但是有可能有编译器把long编译成64位的。保险起见可以加一个8的情况。

File: src/kservice.c [hunk 2]
```diff
@@ -167,16 +168,28 @@ RT_WEAK void *rt_memset(void *s, int c, rt_ubase_t count)
...
+            *(((unsigned char *)&buffer)+3) = d;
+        }
+        else if (LBLOCKSIZE == 2)
         {
-            buffer = 0;
-            for (i = 0; i < LBLOCKSIZE; i ++)
-                buffer = (buffer << 8) | d;
+            *(((unsigned char *)&buffer)+0) = d;
+            *(((unsigned char *)&buffer)+1) = d;
         }
```
Comment (mysterywolf) [inline src/kservice.c:194]:
```suggestion
else
{
    RT_DEBUG_LOG(RT_TRUE, ("rt_memcpy cannot handle this case, please enable RT_KSERVICE_USING_TINY_SIZE\n"));
    ASSERT(0);
}
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: src/kservice.c (hunk 2 above)
Comment (mysterywolf) [reply src/kservice.c:194]: @CuttySark1869 先不要按照此建议修改
@Guozhanxin 郭老师你觉得是else这么写好，还是上来assert LBLOCKSIZE比较好？
```c
RT_ASSERT(LBLOCKSIZE == 2 || LBLOCKSIZE == 4 || LBLOCKSIZE == 8);
```

Comment (Guozhanxin) [reply src/kservice.c:194]: 我感觉都可以吧，

**Review by Guozhanxin — APPROVED** [review]

File: src/kservice.c [hunk 3]
```diff
@@ -152,12 +154,11 @@ RT_WEAK void *rt_memset(void *s, int c, rt_ubase_t count)
 #define UNALIGNED(X)    ((long)X & (LBLOCKSIZE - 1))
 #define TOO_SMALL(LEN)  ((LEN) < LBLOCKSIZE)
```
Comment (mysterywolf) [inline src/kservice.c:156]:
```suggestion
RT_ASSERT(LBLOCKSIZE == 2 || LBLOCKSIZE == 4 || LBLOCKSIZE == 8);
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: src/kservice.c (hunk 3 above)
Comment (mysterywolf) [reply src/kservice.c:156]: 请把这句话加到第156行

Comment (CuttySark1869) [reply src/kservice.c:156]: done

**Review by mysterywolf — APPROVED** [review]

**Comment by mysterywolf** [comment]
已合并 感谢提交PR！

**Pull request merged**